src/utilities/xml/XMLInitializer.cpp

```diff
@@ -0,0 +1,61 @@
+*
+*  Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+*  following conditions are met:
+*
+*  (1) Redistributions of source code must retain the above copyright notice, this list of conditions and the following
+*  disclaimer.
+*
+*  (2) Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following
+*  disclaimer in the documentation and/or other materials provided with the distribution.
+*
+*  (3) Neither the name of the copyright holder nor the names of any contributors may be used to endorse or promote products
+*  derived from this software without specific prior written permission from the respective party.
+*
+*  (4) Other than as required in clauses (1) and (2), distributions in any form of modifications or other derivative works
+*  may not use the "OpenStudio" trademark, "OS", "os", or any other confusingly similar designation without specific prior
+*  written permission from Alliance for Sustainable Energy, LLC.
+*
+*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER(S) AND ANY CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+*  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+*  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE UNITED STATES GOVERNMENT, OR THE UNITED
+*  STATES DEPARTMENT OF ENERGY, NOR ANY OF THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+*  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+*  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+*  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+*  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***********************************************************************************************************************/
+
+#include "XMLInitializer.hpp"
+#include <libxml/parser.h>
+#include <libxslt/transform.h>
+#include <libxslt/xslt.h>
+
+namespace openstudio::detail {
+
+XMLInitializer::XMLInitializer() {
+  // Initialize on the first **Actual** usage
+
+  // init the parser (keeps libxml2 thread safe)
+  xmlInitParser();
+
+  // set some libxml global variables
+  xmlIndentTreeOutput = 1;
+  xmlKeepBlanksDefaultValue = 0;
+  xmlSubstituteEntitiesDefaultValue = 1;
+  xmlLoadExtDtdDefaultValue = 1;
+  xmlDoValidityCheckingDefaultValue = 0;
+
+  xsltInit();
+  xsltSetXIncludeDefault(1);
+}
+
+XMLInitializer::~XMLInitializer() {
+  // This should be done after we're totally done processing stuff...
+  // So that's why it's done by a static object... Note that no cleanup happens in case it was never used, because --reference_counter would be -1
+  xsltCleanupGlobals();
+  xmlCleanupParser();
+}
```

Comment:
We want this to happen at the end of the entire program. In case we use an XMLValidator, then destroy it, and use another one.